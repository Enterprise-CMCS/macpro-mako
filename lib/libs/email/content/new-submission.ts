import { DateTime } from "luxon";
import { Authority, OneMac } from "shared-types";
import {
  CommonVariables,
  formatAttachments,
  formatDate,
  formatNinetyDaysDate,
} from "..";

export const newSubmission = {
  [Authority.MED_SPA]: {
    cms: async (variables: OneMac & CommonVariables) => {
      return {
        subject: `Medicaid SPA ${variables.id} Submitted`,
        html: `
<p>The OneMAC Submission Portal received a Medicaid SPA Submission:</p>
<ul>
<li>The submission can be accessed in the OneMAC application, which you 
can find at <a href='${variables.applicationEndpointUrl}'>this link</a>.</li>
<li>If you are not already logged in, please click the "Login" link 
at the top of the page and log in using your Enterprise User 
Administration (EUA) credentials.</li>
<li>After you have logged in, you will be taken to the OneMAC application. 
The submission will be listed on the dashboard page, and you can view its 
details by clicking on its ID number.</li>
</ul>
<p>
<br><b>State or territory:</b> ${variables.territory}
<br><b>Name:</b> ${variables.submitterName}
<br><b>Email:</b> ${variables.submitterEmail}
<br><b>Medicaid SPA ID: ${variables.id}</b>
<br><b>Proposed Effective Date:</b> ${formatDate(
          variables.notificationMetadata?.proposedEffectiveDate!,
        )}
</p>
<b>Summary:</b>
<br>${variables.additionalInformation}
<br>
<br><b>Files:</b>
<br>${formatAttachments("html", variables.attachments)}
<br>
<p>If the contents of this email seem suspicious, do not open them, and instead 
forward this email to <a href='mailto:SPAM@cms.hhs.gov'>SPAM@cms.hhs.gov</a>.</p>
<p>Thank you!</p>`,
        text: `
The OneMAC Submission Portal received a Medicaid SPA Submission:

The submission can be accessed in the OneMAC application, which you can 
find at ${variables.applicationEndpointUrl}.

If you are not already logged in, please click the "Login" link at the top 
of the page and log in using your Enterprise User Administration (EUA) 
credentials.

After you have logged in, you will be taken to the OneMAC application. 
The submission will be listed on the dashboard page, and you can view its 
details by clicking on its ID number.

State or territory: ${variables.territory}
Name: ${variables.submitterName}
Email: ${variables.submitterEmail}
Medicaid SPA ID: ${variables.id}
Proposed Effective Date: ${formatDate(
          variables.notificationMetadata?.proposedEffectiveDate!,
        )}

Summary:
${variables.additionalInformation}

Files:
${formatAttachments("text", variables.attachments)}

If the contents of this email seem suspicious, do not open them, and instead 
forward this email to SPAM@cms.hhs.gov.

Thank you!`,
      };
    },
    state: async (variables: OneMac & CommonVariables) => {
      return {
        subject: `Your SPA ${variables.id} has been submitted to CMS`,
        html: `
<p>This response confirms that you submitted a Medicaid SPA to CMS for review:</p>
<p>
<br><b>State or territory:</b> ${variables.territory}
<br><b>Name:</b> ${variables.submitterName}
<br><b>Email Address:</b> ${variables.submitterEmail}
<br><b>Medicaid SPA ID: ${variables.id}</b>
<br><b>Proposed Effective Date:</b> ${formatDate(
          variables.notificationMetadata?.proposedEffectiveDate!,
        )}
<br><b>90th Day Deadline:</b> ${formatNinetyDaysDate(
          variables.notificationMetadata?.submissionDate,
        )}
</p>
<b>Summary:</b>
<br>${variables.additionalInformation}
<br>
<p>This response confirms the receipt of your Medicaid State Plan Amendment 
(SPA or your response to a SPA Request for Additional Information (RAI)). 
You can expect a formal response to your submittal to be issued within 90 days, 
before ${formatNinetyDaysDate(
          variables.notificationMetadata?.submissionDate,
        )}.</p>
<p>This mailbox is for the submittal of State Plan Amendments and non-web-based
responses to Requests for Additional Information (RAI) on submitted SPAs only.
Any other correspondence will be disregarded.</p>
<p>If you have questions or did not expect this email, please contact 
<a href='mailto:spa@cms.hhs.gov'>spa@cms.hhs.gov</a>.</p>
<p>Thank you!</p>`,
        text: `
This response confirms that you submitted a Medicaid SPA to CMS for review:

State or territory: ${variables.territory}
Name: ${variables.submitterName}
Email Address: ${variables.submitterEmail}
Medicaid SPA ID: ${variables.id}
Proposed Effective Date:  ${formatDate(
          variables.notificationMetadata?.proposedEffectiveDate!,
        )}
90th Day Deadline: ${formatNinetyDaysDate(
          variables.notificationMetadata?.submissionDate,
        )}

Summary:
${variables.additionalInformation}

This response confirms the receipt of your Medicaid State Plan Amendment 
(SPA or your response to a SPA Request for Additional Information (RAI)). 
You can expect a formal response to your submittal to be issued within 90 days, 
before ${formatNinetyDaysDate(variables.notificationMetadata?.submissionDate)}.

This mailbox is for the submittal of State Plan Amendments and non-web-based
responses to Requests for Additional Information (RAI) on submitted SPAs only.
Any other correspondence will be disregarded.

If you have questions or did not expect this email, please contact 
spa@cms.hhs.gov.

Thank you!`,
      };
    },
  },
  [Authority.CHIP_SPA]: {
    cms: async (variables: OneMac & CommonVariables) => {
      return {
        subject: `New CHIP SPA ${variables.id} Submitted`,
        html: `
<p>The OneMAC Submission Portal received a CHIP State Plan Amendment:</p>
<ul>
<li>The submission can be accessed in the OneMAC Micro application, which 
you can find at <a href='${variables.applicationEndpointUrl}'>
${variables.applicationEndpointUrl}</a>.</li>
<li>If you are not already logged in, please click the "Login" link at the 
top of the page and log in using your Enterprise User Administration (EUA) 
credentials.</li>
<li>After you have logged in, you will be taken to the OneMAC Micro 
application. The submission will be listed on the dashboard page, and you 
can view its details by clicking on its ID number.</li>
</ul>
<p>
<br><b>State or territory:</b> ${variables.territory}
<br><b>Name:</b> ${variables.submitterName}
<br><b>Email:</b> ${variables.submitterEmail}
<br><b>CHIP SPA Package ID:</b> ${variables.id}
</p><br/>
Summary:
<br>${variables.additionalInformation}
<br>
<p>
<br>Files:
<br>${formatAttachments("html", variables.attachments)}
<br></p>
<p>If the contents of this email seem suspicious, do not open them, and instead 
forward this email to <a href='mailto:SPAM@cms.hhs.gov'>SPAM@cms.hhs.gov</a>.</p>
<p>Thank you!</p>`,
        text: `
The OneMAC Submission Portal received a CHIP State Plan Amendment:

- The submission can be accessed in the OneMAC Micro application, which you can 
  find at ${variables.applicationEndpointUrl}.
- If you are not already logged in, please click the "Login" link at the 
  top of the page and log in using your Enterprise User Administration (EUA) 
  credentials.
- After you have logged in, you will be taken to the OneMAC Micro 
  application. The submission will be listed on the dashboard page, and you 
  can view its details by clicking on its ID number.

State or territory: ${variables.territory}
Name: ${variables.submitterName}
Email: ${variables.submitterEmail}
CHIP SPA Package ID: ${variables.id}

Summary:
${variables.additionalInformation}

Files:
${formatAttachments("html", variables.attachments)}

If the contents of this email seem suspicious, do not open them, and instead 
forward this email to SPAM@cms.hhs.gov.

Thank you!`,
      };
    },
    state: async (variables: OneMac & CommonVariables) => {
      return {
        subject: `Your CHIP SPA ${variables.id} has been submitted to CMS`,
        html: `
    <p>This is confirmation that you submitted a CHIP State Plan Amendment 
    to CMS for review:</p>
    <p>
    <br><b>State or territory:</b> ${variables.territory}
    <br><b>Name:</b> ${variables.submitterName}
    <br><b>Email Address:</b> ${variables.submitterEmail}
    <br><b>CHIP SPA Package ID:</b> ${variables.id}
    </p>
    Summary:
    <br>${variables.additionalInformation}
    <br>
    <p>This response confirms the receipt of your CHIP State Plan Amendment 
    (CHIP SPA). You can expect a formal response to your submittal from CMS 
    at a later date.
    </p>
    <p>If you have questions or did not expect this email, please contact 
    <a href='mailto:CHIPSPASubmissionMailBox@CMS.HHS.gov'>CHIPSPASubmissionMailBox@CMS.HHS.gov</a> or your state lead.</p>
    <p>Thank you!</p>`,
        text: `
    This is confirmation that you submitted a CHIP State Plan Amendment 
    to CMS for review:
    
    State or territory: ${variables.territory}
    Name: ${variables.submitterName}
    Email Address: ${variables.submitterEmail}
    CHIP SPA Package ID: ${variables.id}
    
    Summary:
    ${variables.additionalInformation}
    
    This response confirms the receipt of your CHIP State Plan Amendment 
    (CHIP SPA). You can expect a formal response to your submittal from CMS 
    at a later date.
    
    If you have questions or did not expect this email, please contact 
    CHIPSPASubmissionMailBox@CMS.HHS.gov.
    
    Thank you!`,
      };
    },
  },

  [Authority["1915c"]]: {
    cms: async (variables: OneMac & CommonVariables) => {
      return {
        subject: `1915(c) ${variables.id} Submitted`,
        html: `
  <p>The OneMAC Submission Portal received a 1915(c) Appendix K Amendment Submission:</p>
  <ul>
  <li>The submission can be accessed in the OneMAC application, which you
  can find at <a href='${variables.applicationEndpointUrl}'>this link</a>.</li>
  <li>If you are not already logged in, please click the "Login" link
  at the top of the page and log in using your Enterprise User
  Administration (EUA) credentials.</li>
  <li>After you have logged in, you will be taken to the OneMAC application.
  The submission will be listed on the dashboard page, and you can view its
  details by clicking on its ID number.</li>
  </ul>
  <p>
  <br><b>State or territory:</b> ${variables.territory}
  <br><b>Name:</b> ${variables.submitterName}
  <br><b>Email Address:</b> ${variables.submitterEmail}
  <br><b>Amendment Title:</b> ${variables.appkTitle}
  <br><b>Waiver Amendment Number:</b> ${variables.id}
  <br><b>Waiver Authority:</b> 1915(c)
  <br><b>Proposed Effective Date: ${formatDate(
    variables.notificationMetadata?.proposedEffectiveDate!,
  )}
  </p>
  Summary:
  <br>${variables.additionalInformation}
  <br>Files:
  <br>${formatAttachments("html", variables.attachments)}
  <p>If the contents of this email seem suspicious, do not open them, and instead
  forward this email to <a href='mailto:SPAM@cms.hhs.gov'>SPAM@cms.hhs.gov</a>.</p>
  <p>Thank you!</p>`,
        text: `
  This response confirms the submission of your [insert Waiver Action] to CMS for review:

  State or territory: ${variables.territory}
  Name: ${variables.submitterName}
  Email Address: ${variables.submitterEmail}
  Amendment Title: ${variables.appkTitle}
  Waiver Amendment Number: ${variables.id}
  Waiver Authority: 1915(c)
  Proposed Effective Date: ${formatDate(
    variables.notificationMetadata?.proposedEffectiveDate!,
  )}

  Summary:
  ${variables.additionalInformation}

  Files:
  ${formatAttachments("html", variables.attachments)}

  If the contents of this email seem suspicious, do not open them, and instead forward this email to SPAM@CMS.HHS.gov

  Thank you!
  `,
      };
    },
    state: async (variables: OneMac & CommonVariables) => {
      return {
        subject: `Your 1915(c) ${variables.id} has been submitted to CMS`,
        html: `
  <p>This response confirms the submission of your 1915(c) Waiver to CMS for review:</p>
  <p>
  <br><b>State or territory:</b> ${variables.territory}
  <br><b>Name:</b> ${variables.submitterName}
  <br><b>Email Address:</b> ${variables.submitterEmail}
  <br><b>Initial Waiver Number:</b> ${variables.id}
  <br><b>Waiver Authority:</b> 1915(c)
  <br><b>Proposed Effective Date:</b> ${formatDate(
    variables.notificationMetadata?.proposedEffectiveDate!,
  )}
  <br><b>90th Day Deadline:</b> ${formatNinetyDaysDate(
    variables.notificationMetadata?.submissionDate,
  )}
  </p>
  Summary:
  <br>${variables.additionalInformation}
  <p>
  This response confirms the receipt of your Waiver request or your response
  to a Waiver Request for Additional Information (RAI). You can expect a formal
  response to your submittal to be issued within 90 days, before ${formatNinetyDaysDate(
    variables.notificationMetadata?.submissionDate,
  )}.
  </p>
  <p>
  This mailbox is for the submittal of Section 1915(b) and 1915(c) Waivers,
  responses to Requests for Additional Information (RAI) on Waivers, and
  extension requests on Waivers only.  Any other correspondence will be disregarded.
  </p>
  <p>If you have questions, please contact
  <a href='mailto:spa@cms.hhs.gov'>spa@cms.hhs.gov</a> or your state lead.</p>
  <p>Thank you!</p>`,
        text: `
  This response confirms the submission of your 1915(c) Waiver to CMS for review:

  State or territory:${variables.territory}
  Name: ${variables.submitterName}
  Email Address: ${variables.submitterEmail}
  Initial Waiver Number: ${variables.id}
  Waiver Authority: 1915(c)
  Proposed Effective Date: ${formatDate(
    variables.notificationMetadata?.proposedEffectiveDate!,
  )}
  90th Day Deadline: ${formatNinetyDaysDate(
    variables.notificationMetadata?.submissionDate,
  )}

  Summary:
  ${variables.additionalInformation}

  This response confirms the receipt of your Waiver request or your response
  to a Waiver Request for Additional Information (RAI). You can expect a formal
  response to your submittal to be issued within 90 days, before ${formatNinetyDaysDate(
    variables.notificationMetadata?.submissionDate,
  )}.

  This mailbox is for the submittal of Section 1915(b) and 1915(c) Waivers,
  responses to Requests for Additional Information (RAI) on Waivers, and
  extension requests on Waivers only.  Any other correspondence will be disregarded.

  If you have questions, please contact SPA@cms.hhs.gov or your state lead.

  Thank you!`,
      };
    },
  },
};
