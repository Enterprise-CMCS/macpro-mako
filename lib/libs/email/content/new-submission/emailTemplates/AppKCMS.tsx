import { emailTemplateValue } from "../data";
import { Events, CommonEmailVariables } from "shared-types";
import { DateTime } from "luxon";
import {
  Html,
  Container,
  Heading,
  Hr,
  Section,
  Body,
  Head,
  Preview,
} from "@react-email/components";
import {
  LoginInstructions,
  PackageDetails,
  SpamWarning,
  styles,
  EmailNav,
} from "../../email-components";

// 1915c - app K
export const AppKCMSEmail = (props: {
  variables: Events["NewAppKSubmission"] & CommonEmailVariables;
}) => {
  const variables = props.variables;
  const previewText = `Appendix K Amendment Submitted `;
  return (
    <Html>
      <Head />
      <Preview>{previewText}</Preview>
      <Body style={styles.main}>
        <Container style={styles.container}>
          <EmailNav appEndpointUrl={variables.applicationEndpointUrl} />
          <Section style={styles.upperSection}>
            <Heading style={styles.h1}>
              The OneMAC Submission Portal received a 1915(c) Appendix K
              Amendment Submission:
            </Heading>
            <Hr style={styles.divider} />
            <LoginInstructions
              appEndpointURL={variables.applicationEndpointUrl}
            />
            <PackageDetails
              details={{
                "State or territory": variables.territory,
                Name: variables.submitterName,
                "Email Address": variables.submitterEmail,
                "Amendment Title": variables.title ?? null,
                "Waiver Amendment Number": variables.id,
                "Waiver Authority": variables.seaActionType,
                "Proposed Effective Date": DateTime.fromMillis(
                  Number(variables.proposedEffectiveDate),
                ).toFormat("DDDD"),
                Summary: variables.additionalInformation,
              }}
              attachments={variables.attachments}
            />
          </Section>
          <SpamWarning />
        </Container>
      </Body>
    </Html>
  );
};

// To preview with on 'email-dev'
const AppKCMSEmailPreview = () => {
  return (
    <AppKCMSEmail
      variables={{
        ...emailTemplateValue,
        seaActionType: "Amend",
        title: "Title",
        waiverIds: ["123"],
        state: "State",
        origin: "mako",
      }}
    />
  );
};

export default AppKCMSEmailPreview;
