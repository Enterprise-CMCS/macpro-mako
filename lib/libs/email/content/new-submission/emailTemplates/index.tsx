export { MedSpaCMSEmail } from "./MedSpaCMS";
export { MedSpaStateEmail } from "./MedSpaState";
export { ChipSpaCMSEmail } from "./ChipSpaCMS";
export { ChipSpaStateEmail } from "./ChipSpaState";
export { Waiver1915bCMSEmail } from "./Waiver1915bCMS";
export { Waiver1915bStateEmail } from "./Waiver1915bState";
export { AppKCMSEmail } from "./AppKCMS";
export { AppKStateEmail } from "./AppKState";
