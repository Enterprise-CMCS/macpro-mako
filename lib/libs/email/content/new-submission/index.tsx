import * as React from "react";
import { Authority, EmailAddresses, OneMac } from "shared-types";
import { CommonVariables } from "../..";
import {
  MedSpaCMSEmail,
  MedSpaStateEmail,
  ChipSpaCMSEmail,
  ChipSpaStateEmail,
  Waiver1915bCMSEmail,
  Waiver1915bStateEmail,
  AppKCMSEmail,
  AppKStateEmail,
} from "./emailTemplate";
import { render } from "@react-email/components";

export const newSubmission = {
  [Authority.MED_SPA]: {
    cms: async (
      variables: OneMac &
        CommonVariables & { emails: EmailAddresses } & {
          emails: EmailAddresses;
        },
    ) => {
      return {
        to: variables.emails.osgEmail,
        subject: `Medicaid SPA ${variables.id} Submitted`,
        html: render(<MedSpaCMSEmail variables={variables} />, {
          pretty: true,
        }),
        text: render(<MedSpaCMSEmail variables={variables} />, {
          plainText: true,
        }),
      };
    },
    state: async (
      variables: OneMac & CommonVariables & { emails: EmailAddresses },
    ) => {
      return {
        to: `"${variables.submitterName}" <${variables.submitterEmail}>`,
        subject: `Your SPA ${variables.id} has been submitted to CMS`,
        html: render(<MedSpaStateEmail variables={variables} />, {
          pretty: true,
        }),
        text: render(<MedSpaStateEmail variables={variables} />, {
          plainText: true,
        }),
      };
    },
  },
  [Authority.CHIP_SPA]: {
    cms: async (
      variables: OneMac & CommonVariables & { emails: EmailAddresses },
    ) => {
      return {
        to: variables.emails.chipInbox,
        cc: variables.emails.chipCcList,
        subject: `New CHIP SPA ${variables.id} Submitted`,
        html: render(<ChipSpaCMSEmail variables={variables} />, {
          pretty: true,
        }),
        text: render(<ChipSpaCMSEmail variables={variables} />, {
          plainText: true,
        }),
      };
    },
    state: async (
      variables: OneMac & CommonVariables & { emails: EmailAddresses },
    ) => {
      return {
        to: `"${variables.submitterName}" <${variables.submitterEmail}>`,
        subject: `Your CHIP SPA ${variables.id} has been submitted to CMS`,
        html: render(<ChipSpaStateEmail variables={variables} />, {
          pretty: true,
        }),
        text: render(<ChipSpaStateEmail variables={variables} />, {
          plainText: true,
        }),
      };
    },
  },
  [Authority["1915b"]]: {
    cms: async (
      variables: OneMac & CommonVariables & { emails: EmailAddresses },
    ) => {
      return {
        to: variables.emails.osgEmail,
        subject: `${variables.authority} ${variables.id} Submitted`,
        html: render(<Waiver1915bCMSEmail variables={variables} />, {
          pretty: true,
        }),
        text: render(<Waiver1915bCMSEmail variables={variables} />, {
          plainText: true,
        }),
      };
    },
    state: async (
      variables: OneMac & CommonVariables & { emails: EmailAddresses },
    ) => {
      return {
        to: `"${variables.submitterName}" <${variables.submitterEmail}>`,
        subject: `Your ${variables.actionType} ${variables.id} has been submitted to CMS`,
        html: render(<Waiver1915bStateEmail variables={variables} />, {
          pretty: true,
        }),
        text: render(<Waiver1915bStateEmail variables={variables} />, {
          plainText: true,
        }),
      };
    },
  },
  [Authority["1915c"]]: {
    cms: async (
      variables: OneMac & CommonVariables & { emails: EmailAddresses },
    ) => {
      return {
        to: variables.emails.osgEmail,
        subject: `1915(c) ${variables.id} Submitted`,
        html: render(<AppKCMSEmail variables={variables} />, {
          pretty: true,
        }),
        text: render(<AppKCMSEmail variables={variables} />, {
          plainText: true,
        }),
      };
    },
    state: async (
      variables: OneMac & CommonVariables & { emails: EmailAddresses },
    ) => {
      return {
        to: `"${variables.submitterName}" <${variables.submitterEmail}>`,
        subject: `Your 1915(c) ${variables.id} has been submitted to CMS`,
        html: render(<AppKStateEmail variables={variables} />, {
          pretty: true,
        }),
        text: render(<AppKStateEmail variables={variables} />, {
          plainText: true,
        }),
      };
    },
  },
};
