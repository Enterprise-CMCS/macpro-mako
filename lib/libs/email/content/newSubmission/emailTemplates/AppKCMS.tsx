import { CommonEmailVariables, Events } from "shared-types";
import { formatDate } from "shared-utils";

import {
  Attachments,
  BasicFooter,
  Divider,
  LoginInstructions,
  PackageDetails,
} from "../../email-components";
import { BaseEmailTemplate } from "../../email-templates";

type AppKEmailProps = Events["AppKSubmission"] & CommonEmailVariables;

// 1915c - app K
export const AppKCMSEmail = ({ variables }: { variables: AppKEmailProps }) => {
  return (
    <BaseEmailTemplate
      previewText="Appendix K Amendment Submitted"
      heading="The OneMAC Submission Portal received a 1915(c) Appendix K Amendment Submission:"
      applicationEndpointUrl={variables.applicationEndpointUrl}
      footerContent={<BasicFooter />}
    >
      <Divider />
      <LoginInstructions appEndpointURL={variables.applicationEndpointUrl} useThisLink />
      <PackageDetails
        details={{
          "State or Territory": variables.territory,
          Name: variables.submitterName,
          "Email Address": variables.submitterEmail,
          "Amendment Title": variables.title ?? null,
          "Waiver Amendment Number": variables.id,
          "Waiver Authority": variables.authority,
          "Proposed Effective Date": formatDate(variables.proposedEffectiveDate),
          Summary: variables.additionalInformation,
        }}
      />
      <Attachments attachments={variables.attachments} />
    </BaseEmailTemplate>
  );
};
