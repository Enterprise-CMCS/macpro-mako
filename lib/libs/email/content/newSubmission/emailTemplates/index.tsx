export { MedSpaCMSEmail } from "./MedSpaCMS";
export { MedSpaStateEmail } from "./MedSpaState";
export { ChipSpaCMSEmail } from "./ChipSpaCMS";
export { ChipSpaStateEmail } from "./ChipSpaState";
export { ChipSpaDetailsCMSEmail } from "./ChipSpaDetailsCMS";
export { ChipSpaDetailsStateEmail } from "./ChipSpaDetailsState";
export { AppKCMSEmail } from "./AppKCMS";
export { AppKStateEmail } from "./AppKState";
export { Waiver1915bCMSEmail } from "./Waiver1915bCMS";
export { Waiver1915bStateEmail } from "./Waiver1915bState";
