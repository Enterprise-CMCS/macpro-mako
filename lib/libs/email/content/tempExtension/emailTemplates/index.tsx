export { TempExtCMSEmail } from "./TempExtCMS";
export { TempExtStateEmail } from "./TempExtState";
