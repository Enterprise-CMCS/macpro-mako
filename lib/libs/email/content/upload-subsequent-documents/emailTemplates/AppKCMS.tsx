import { CommonEmailVariables, Events } from "shared-types";
import {
  PackageDetails,
  BasicFooter,
  Attachments,
  SubDocHowToAccess,
  Divider,
} from "../../email-components";
import { BaseEmailTemplate } from "../../email-templates";
import { styles } from "../../email-styles";
import { Text } from "@react-email/components";

export const AppKCMSEmail = (props: {
  variables: Events["UploadSubsequentDocuments"] & CommonEmailVariables;
}) => {
  const variables = props.variables;
  const previewText = `Action required: review new documents for 1915(c) ${variables.id} in OneMAC.`;
  const heading = `New documents have been submitted for 1915(c) ${variables.id} in OneMAC.`;
  return (
    <BaseEmailTemplate
      previewText={previewText}
      heading={heading}
      applicationEndpointUrl={variables.applicationEndpointUrl}
      footerContent={<BasicFooter />}
    >
      <PackageDetails
        details={{
          "State or territory": variables.territory,
          "1915(c) Appendix K ID": variables.id,
          Summary: variables.additionalInformation,
        }}
      />
      <Attachments attachments={variables.attachments} />
      <Divider />
      <SubDocHowToAccess appEndpointURL={variables.applicationEndpointUrl} />
      <Text style={{ ...styles.text.base, marginTop: "16px" }}>Thank you.</Text>
    </BaseEmailTemplate>
  );
};
