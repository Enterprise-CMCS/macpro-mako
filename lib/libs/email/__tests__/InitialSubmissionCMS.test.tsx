import { describe, it, expect } from "vitest";
import { render } from "@testing-library/react";
import { AppKCMSEmail } from "../content/new-submission/emailTemplates/AppKCMS";
import { emailTemplateValue } from "../mock-data/new-submission";
import ChipSpaCMSEmailPreview from "../preview/Initial_Submissions/CMS/CHIP_SPA";
import MedicaidSPA from "../preview/Initial_Submissions/CMS/Medicaid_SPA";
import TempExtCMSPreview from "../preview/Initial_Submissions/CMS/Temp_Extension";
import Waiver1915bCMSEmailPreview from "../preview/Initial_Submissions/CMS/Waiver_Capitated";

describe("Initial Submission CMS Email Snapshot Test", () => {
  it("renders a AppkCMSEmail Preview Template", async () => {
    const template = (
      <AppKCMSEmail
        variables={{
          ...emailTemplateValue,
          event: "app-k",
          id: "CO-1234.R21.00",
          authority: "1915(c)",
          actionType: "Amend",
          title: "A Perfect Appendix K Amendment Title",
          attachments: {
            appk: {
              label: "AppK",
              files: [],
            },
            other: {
              label: "Other",
              files: [],
            },
          },
        }}
      />
    );

    expect(template).toMatchSnapshot();
  });
  it("renders a Chipspa Preview Template", () => {
    const template = render(<ChipSpaCMSEmailPreview />);

    expect(template).toMatchSnapshot();
  });
  it("renders a Medicaid Spa Preview Template", () => {
    const template = render(<MedicaidSPA />);

    expect(template).toMatchSnapshot();
  });
  it("renders a TempExt Preview Template", () => {
    const template = render(<TempExtCMSPreview />);

    expect(template).toMatchSnapshot();
  });
  it("renders a Waiver Capitated Preview Template", () => {
    const template = render(<Waiver1915bCMSEmailPreview />);

    expect(template).toMatchSnapshot();
  });
});
