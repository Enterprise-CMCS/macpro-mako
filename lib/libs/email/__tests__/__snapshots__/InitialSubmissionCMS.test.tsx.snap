// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Initial Submission CMS Email Snapshot Test > renders a AppkCMSEmail Preview Template 1`] = `
<UNDEFINED
  variables={
    {
      "additionalInformation": "This submission includes necessary documentation for requested updates to the stateâ€™s Medicaid plan, in alignment with CMS requirements.",
      "appkParentId": null,
      "applicationEndpointUrl": "https://mako-dev.cms.gov/",
      "origin": "mako",
      "proposedEffectiveDate": 1728495789885,
      "submitterEmail": "george@example.com",
      "submitterName": "George Harrison",
      "territory": "CO",
      "timestamp": 1735333827084,
    }
  }
/>
`;
